coalib/io/FileFactory.py

```diff
@@ -0,0 +1,127 @@
+import os
+
+from coala_utils.decorators import generate_eq
+
+
+@generate_eq('name', 'timestamp')
+class FileFactory:
+    """
+    The ``FileFactory`` is a wrapper class whose objects contain the
+    following information about the file:
+
+        * The filename (absolute file path).
+        * The modification timestamp of the file.
+        * The file in raw format.
+        * The file in the form of a string UTF-8 decoded from the raw format.
+        * The lines of the file in the form of a tuple.
+
+    The equality of the object is checked just based on its filename and
+    timestamp.
```

Comment:
--> Two `FileFactory` objects are conisdered equal if they have the same filename and timestamp.